packages/app/src/systems/Error/components/Providers/ReportErrorProvider.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { createContext, useEffect } from 'react';
+
+import { db } from '../../../Core';
+import { errorToFuelError } from '../../utils';
+
+const initialState = {};
+
+const FuelErrorContext = createContext(initialState);
+const { Provider } = FuelErrorContext;
+
+type ErrorProviderProps = {
+  children: React.ReactNode;
+};
+
+const ErrorProvider = ({ children }: ErrorProviderProps) => {
+  const handleError = async (e: ErrorEvent) => {
+    const error = errorToFuelError(e.error);
+    await db.errors.add(error);
+    // return false to allow the default handler to run.
+    return false;
+  };
+
+  const handleUnhandledRejection = async (e: PromiseRejectionEvent) => {
+    e.preventDefault();
+    const error = errorToFuelError(e.reason);
+    await db.errors.add(error);
+    // return false to allow the default handler to run.
+    return true;
+  };
```

Comment:
will this errors be reported? I don't see anything connecting it to the reportErrorMachine, so it won't update the `hasErrors` context

RN as I'm understanding this will be only saved in db errors, and will be reported later if user has another error somewhere